core/repository/http/src/main/java/org/eclipse/rdf4j/repository/http/HTTPRepository.java

```diff
@@ -111,49 +113,76 @@ public HTTPRepository(final String repositoryURL) {
 	 * ---------------* public methods * ---------------
 	 */
 
+	@Override
 	public void setDataDir(final File dataDir) {
 		this.dataDir = dataDir;
 	}
 
+	@Override
 	public File getDataDir() {
 		return dataDir;
 	}
 
-	public synchronized SesameClient getSesameClient() {
-		if (client == null) {
-			client = dependentClient = new SesameClientImpl();
+	@Override
+	public SesameClient getSesameClient() {
+		SesameClient result = client;
+		if (result == null) {
+			synchronized (this) {
+				result = client;
+				if (result == null) {
+					result = client = dependentClient = new SesameClientImpl();
```

Comment:
It has proven performance advantages by reducing volatile reads on the variable reference. I agree though, Java should have a better mechanism for caching volatile reads to avoid it.